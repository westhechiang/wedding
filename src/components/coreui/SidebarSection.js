import styled from 'styled-components';

const width = {
  fullWidth: '100%',
  inherit: 'inherit',
  undefined: 'auto',
};

const SidebarSection = styled.div`
  height: ${props => props.height || 'auto'};
  width: ${props => width[props.width]};
  display: flex;
  overflow-y: ${props => props.overflowY || 'auto'};
  overflow-x: hidden;
  flex-direction: column;
  justify-content: flex-start;
  position: relative;
`;

export default SidebarSection;
