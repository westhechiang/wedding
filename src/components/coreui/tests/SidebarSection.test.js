import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SidebarSection from '../../../components/coreui/SidebarSection';

describe('<SidebarSection />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<SidebarSection />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "fullWidth" snapshot', () => {
    const wrapper = shallow(<SidebarSection width="fullWidth" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "inherit" snapshot', () => {
    const wrapper = shallow(<SidebarSection width="inherit" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its snapshot when passed a "height" prop', () => {
    const wrapper = shallow(<SidebarSection height="50px" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
