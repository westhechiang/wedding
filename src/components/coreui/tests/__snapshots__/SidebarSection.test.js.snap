// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarSection /> should match its "fullWidth" snapshot 1`] = `
<div
  className="sc-bdVaJa czyqHW"
  width="fullWidth"
/>
`;

exports[`<SidebarSection /> should match its "inherit" snapshot 1`] = `
<div
  className="sc-bdVaJa koPwjx"
  width="inherit"
/>
`;

exports[`<SidebarSection /> should match its empty snapshot 1`] = `
<div
  className="sc-bdVaJa hUpMWE"
/>
`;

exports[`<SidebarSection /> should match its snapshot when passed a "height" prop 1`] = `
<div
  className="sc-bdVaJa YYVmh"
  height="50px"
/>
`;
